setup.sh

```diff
@@ -7,10 +7,18 @@ IFS=$'\n\t'
 CPUS=$(getconf _NPROCESSORS_ONLN 2>/dev/null || getconf NPROCESSORS_ONLN 2>/dev/null || echo 1)
 
 echo "Installing dependencies"
-sudo apt-get update --yes
-sudo apt-get upgrade --yes
-sudo apt-get install cmake doxygen graphviz gcovr llvm g++ pkg-config m4 --yes
-sudo curl --proto '=https' --tlsv1.3 https://sh.rustup.rs -sSf | sh -s -- -y #install rust compiler
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then # Linux
+  sudo apt-get update --yes
+  sudo apt-get install cmake doxygen graphviz gcovr llvm g++ pkg-config m4 wget --yes
+elif [[ "$OSTYPE" == "darwin"* ]]; then # macOS
+  brew update
+  brew install cmake doxygen graphviz pkg-config wget coreutils # `coreutils` installs the `realpath` command
+  brew install --ignore-dependencies gcovr # don't install extra python@3.11, which causes issues for the mozilla build system, see https://bugzilla.mozilla.org/show_bug.cgi?id=1766497
+else
+  echo "Unsupported OS"
+  exit 1
+fi
+sudo curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh -s -- -y #install rust compiler
```

Comment:
I was thinking about this and I think it might feel better to have two separate scripts: `install_dependencies.sh` and `build_spidermonkey.sh`. 

Some users are weary of dependency installation scripts and would prefer to run them themselves.  What do you think about this @Xmader

Reply:
Will do in a separate PR.